api.py

```diff
@@ -65,10 +70,18 @@ def init_rollbar():
         # send exceptions from `app` to rollbar, using flask's signal system.
         got_request_exception.connect(rollbar.contrib.flask.report_exception, app)
 
+
 @app.before_request
 def validate_api_key():
-    if not request.args.get('apikey', None) == app.config['APIKEY']:
-        abort(403)
+    apikey = request.args.get('apikey', request.headers.get('X-Api-Key', None)) # get the API key from a request param or a header
+
+    sql = get_table('Sesh_RMC_Account').select().where(sqlalchemy.text('API_KEY = :k')).limit(1)
+    result = app.engine.execute(sql, k=apikey)
```

Comment:
that is a very basic SQL requests on a table with not many entries. Add an index on the API_KEY field and we'll be good. My python code will die before anything.